src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,143 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.InvalidDataException;
+import core.basesyntax.exceptions.UserAlreadyExistsException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private User validUser1;
+    private User validUser2;
+    private final RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+    private final StorageDaoImpl storageDao = new StorageDaoImpl();
+
+    @BeforeEach
+    void setUp() {
+        validUser1 = new User();
+        validUser2 = new User();
+
+        validUser1.setLogin("validUser1");
+        validUser1.setPassword("validPassword1");
+        validUser1.setAge(21);
```

Comment:
set MIN_AGE for one. We should test **edge cases**
If **MIN_AGE = 18** and you test only 15 and 21 - it means for age 17 or 18 code can be incorrect and you tests can not find it